pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -177,6 +213,61 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, pa *pav1alpha1.PodAutosc
 	return computeStatus(pa, pc)
 }
 
+func (c *Reconciler) markPodsForRemoval(ctx context.Context, removalCandidates []string, pa *pav1alpha1.PodAutoscaler, want, readyCount int32, pc resourceutil.EndpointsCounter) error {
+	logger := logging.FromContext(ctx)
+
+	pods, err := c.podsLister.Pods(pa.Namespace).List(labels.SelectorFromSet(labels.Set{serving.RevisionUID: pa.Labels[serving.RevisionUID]}))
+	if err != nil {
+		return err
+	}
+
+	podsMap := make(map[string]*corev1.Pod, len(pods))
+	for _, p := range pods {
+		podsMap[p.Name] = p
+	}
+
+	logger.Infof("readyCount = %d, want = %d, removalCandidates = %d", readyCount, want, len(removalCandidates))
+	for _, podName := range removalCandidates {
+		p := podsMap[podName]
+		newP := p.DeepCopy()
+		newP.ObjectMeta.Labels[autoscaling.PreferForScaleDownLabelKey] = "true"
+		patch, err := duck.CreatePatch(p, newP)
+		if err != nil {
+			return err
+		}
+
+		patchBytes, err := patch.MarshalJSON()
+		if err != nil {
+			return err
+		}
+
+		if _, err = c.KubeClientSet.CoreV1().Pods(pa.Namespace).Patch(p.ObjectMeta.Name, types.JSONPatchType, patchBytes); err != nil {
+			return err
+		}
+	}
+
+	// wait for endpoint counts to converge before scaling down
+	ctx, _ = context.WithTimeout(ctx, EndpointConvergenceTimeout)
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("failed to converge endpoints after one minute")
```

Comment:
The "one minute" should probably not be hardcoded if you use a different constant above.